CppSamples/DisplayInformation/ShowGrid/ShowGrid.cpp

```diff
@@ -0,0 +1,276 @@
+#include "SpatialReference.h"
+#include "Surface.h"
+#include "TextSymbol.h"
+#include "USNGGrid.h"
+#include "UTMGrid.h"
+#include "Viewpoint.h"
+
+#include <QFuture>
+
+using namespace Esri::ArcGISRuntime;
+
+ShowGrid::ShowGrid(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(BasemapStyle::ArcGISImagery, this)),
+  m_scene(new Scene(BasemapStyle::ArcGISImagery, this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+
+  // Set the sample's initial GeoView type to MapView
+  m_currentViewType = s_mapView;
+
+  // Set the initial viewpoint of the map
+  constexpr double targetScale = 6450785;
+  m_map->setInitialViewpoint(Viewpoint(Point(-10336141.70018318, 5418213.05332071, SpatialReference::webMercator()), targetScale));
+
+  // Set the intial grid to LatitudeLongitudeGrid
+  m_grid = new LatitudeLongitudeGrid(this);
+}
+
+ShowGrid::~ShowGrid() = default;
+
+void ShowGrid::init()
+{
+  // Register classes for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<ShowGrid>("Esri.Samples", 1, 0, "ShowGridSample");
+}
+
+MapQuickView* ShowGrid::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ShowGrid::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->setGrid(m_grid);
+
+  emit mapViewChanged();
+}
+
+SceneQuickView* ShowGrid::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void ShowGrid::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+    return;
+
+  m_sceneView = sceneView;
+  m_sceneView->setArcGISScene(m_scene);
+
+  // Set the SceneView initially to be invisible
+  m_sceneView->setVisible(false);
+
+  emit sceneViewChanged();
+}
+
+
+void ShowGrid::setViewType(const QString& viewType)
+{
+  if (m_currentViewType == viewType)
+    return;
+
+  m_currentViewType = viewType;
+
+  if (viewType == s_mapView)
+  {
+    m_mapView->setViewpointAsync(m_sceneView->currentViewpoint(ViewpointType::BoundingGeometry), 0);
+
+    m_mapView->setVisible(true);
+    m_sceneView->setVisible(false);
+
+    m_sceneView->setGrid(nullptr);
+    m_mapView->setGrid(m_grid);
+  }
+  else if (viewType == s_sceneView)
+  {
+    m_sceneView->setViewpointAsync(m_mapView->currentViewpoint(ViewpointType::BoundingGeometry), 0);
+
+    m_mapView->setVisible(false);
+    m_sceneView->setVisible(true);
+
+    m_mapView->setGrid(nullptr);
+    m_sceneView->setGrid(m_grid);
+  }
```

Comment:
I haven't built or tested this code, but I was thinking of something conceptually like this
```suggestion
  auto geoView = viewType == s_mapView ? m_mapView : m_sceneView;
  geoView->setViewpointAsync(geoView->currentViewpoint(ViewpointType::BoundingGeometry), 0);

  geoView->setVisible(true);
  geoView->setVisible(false);

  geoView->setGrid(nullptr);
  geoView->setGrid(m_grid);  
```

Reply:
GeoView doesn't have a `setVisible` method - that's only exposed to MapView and SceneView specifically. Otherwise I'd be able to do something like this:

```cpp
  auto geoView = viewType == s_mapView ? m_mapView : m_sceneView;
  auto oldGeoView = viewType != s_mapView ? m_mapView : m_sceneView;
  geoView->setViewpointAsync(oldGeoView->currentViewpoint(ViewpointType::BoundingGeometry), 0);
  geoView->setVisible(true);
  oldGeoView->setVisible(false);
  oldGeoView->setGrid(nullptr);
  geoView->setGrid(m_grid); 
  ```

Reply:
@tanneryould your call on which is better, just seemed like there was an opportunity to streamline

Reply:
While I agree there might be a clever way to streamline it and that would be cool, I think that's a bit out of scope for this sample. I'd likely have to rewrite a bit of it and maybe somehow control each view's visibility on the qml side.

Reply:
Solved, thank you!